### PR TITLE
Fixed issue #09056: Plugin settings password shown in HTML source

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue [security] #09056: Plugin settings password shown in HTML source
Dev: Add Keep password checkbox
Dev: send an array with keepit => true
Dev: by optin to be sure to don't break any existing plugin

### CODE REVIEW

**Comment by Shnoulle** [comment]
```
        'checkPasswordNew'=>array(
            'type'=>'password',
            'label' => 'A new password',
            'default' => '',
            'usekeepit' => true,
        ),
```


194817853-76e8985e-53c8-42ab-a8ef-163142477ff3

**Review by gabrieljenik — COMMENTED** [review]
Sorry, comments were left as "pending" and didn't realize.

**Comment by Shnoulle** [comment]
I review to use same system than https://github.com/LimeSurvey/LimeSurvey/commit/aef109080574bbc4db0631233aaa14fa681935e3

Let me redo whole fix

File: application/controllers/admin/Database.php [hunk 1]
```diff
@@ -657,6 +657,15 @@ private function actionUpdateSurveyLocaleSettings($iSurveyID)
             // Save plugin settings : actually leave it before saving core : we are sure core settings is saved in LS way.
             $pluginSettings = App()->request->getPost('plugin', array());
             foreach ($pluginSettings as $plugin => $settings) {
+                $oPlugin = App()->getPluginManager()->loadPlugin($plugin, null, false);// must be already loaded, do not force reload
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/controllers/admin/Database.php:660]: > Should we do this only for the password plugin, right?
 
I don't know , why not allow if for all settings if $metatData['usekeepit'] ?

File: application/controllers/admin/Database.php [hunk 2]
```diff
@@ -657,6 +657,15 @@ private function actionUpdateSurveyLocaleSettings($iSurveyID)
...
+                if ($oPlugin) {
+                    /* Fix settings for keepit (password) **/
+                    foreach ($settings as $setting => $value) {
+                        if (is_array($value) && !empty($value['keepit'])) {
+                            $settings[$setting] = $oPlugin->get($setting, 'Survey', $iSurveyID, null);
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/controllers/admin/Database.php:665]: I update to don't use array of value, then it fix this issue.

**Comment by Shnoulle** [comment]
Noit the good way, must check metadata only

**Pull request closed without merging**